dhis-2/dhis-services/dhis-service-core/src/main/java/org/hisp/dhis/trackedentity/hibernate/HibernateTrackedEntityInstanceStore.java

```diff
@@ -484,11 +487,90 @@ private String getQuerySelect( TrackedEntityInstanceQueryParams params, boolean
         {
             select.append( ", TEI.trackedentityinstanceid AS teiid" );
         }
+
+        select.append( getOrderParamForSelectClause( params, select ) );
+
         select.append( SPACE );
 
         return select.toString();
     }
 
+    /**
+     * Gets all the static fields from the order parameters and appends them to
+     * the string builder, so they can be used in the query select clause
+     *
+     * @param params
+     * @param select current select statement
+     * @return all the static fields that are not yet present in the select
+     *         statement
+     */
+    private String getOrderParamForSelectClause( TrackedEntityInstanceQueryParams params, StringBuilder select )
+    {
+        StringBuilder orderQuery = new StringBuilder();
+
+        for ( OrderParam orderParam : params.getOrders() )
+        {
+            if ( isStaticColumn( orderParam.getField() ) )
+            {
+                if ( ENROLLED_AT.isPropertyEqualTo( orderParam.getField() ) )
+                {
+                    //In the main query, we need to use the main query alias to fetch the value of the field enrolledAt
+                    orderQuery
+                        .append( addStaticColumn( orderParam.getField(), select.toString(), MAIN_QUERY_ALIAS ) );
+                }
+                else
+                {
+                    orderQuery.append( addStaticColumn( orderParam.getField(), select.toString() ) );
+                }
+            }
+        }
+
+        return orderQuery.toString();
+    }
+
+    /**
+     * Checks if a field is a valid column and whether is present in the current
+     * query.
+     *
+     * @param field name of the field to add in the query
+     * @param sqlQuery current sql query
+     * @return the mapped column name if not present in the query, empty
+     *         otherwise
+     */
+    private String addStaticColumn( String field, String sqlQuery )
```

Comment:
This method is confusing me 😅 

The javadocs say `Checks if a field is a valid column and whether is present in the current query.`. The method returns a `String` and the method name is `addStaticColumn` suggesting that something is added to something. A method that just checking something should probably return a boolean. It should also not have a word like `add` suggesting something is changed. Reminds me of https://en.wikipedia.org/wiki/Command%E2%80%93query_separation